Tests/SwiftGenKitTests/FilterTests.swift

```diff
@@ -0,0 +1,49 @@
+//
+// SwiftGenKit UnitTests
+// Copyright © 2019 SwiftGen
+// MIT Licence
+//
+
+import PathKit
+@testable import SwiftGenKit
+import XCTest
+
+class FilterTests: XCTestCase {
+  func testBasicRegex() throws {
+    let filter = try Filter(pattern: ".+\\.strings$")
+
+    XCTAssertTrue(Path("foo.strings").matches(filter: filter))
+    XCTAssertTrue(Path("Bar.strings").matches(filter: filter))
+    XCTAssertTrue(Path("A/B/C/baz.strings").matches(filter: filter))
+
+    XCTAssertFalse(Path("foo.Strings").matches(filter: filter))
+    XCTAssertFalse(Path("foo.stringsdict").matches(filter: filter))
+  }
+
+  func testCaseSensitive() throws {
+    let filter = try Filter(pattern: ".+\\.strings$")
+
+    XCTAssertTrue(Path("foo.strings").matches(filter: filter))
+    XCTAssertTrue(Path("Bar.strings").matches(filter: filter))
+    XCTAssertFalse(Path("foo.Strings").matches(filter: filter))
+    XCTAssertFalse(Path("Bar.STRINGS").matches(filter: filter))
+  }
+
+  func testCaseInsensitive() throws {
+    let filter = try Filter(pattern: ".+\\.(?i:json)$")
+
+    XCTAssertTrue(Path("foo.json").matches(filter: filter))
+    XCTAssertTrue(Path("foo.Json").matches(filter: filter))
+    XCTAssertTrue(Path("foo.JSON").matches(filter: filter))
+  }
+
+  func testComplex() throws {
+    // match json files with the filename beginning with "SG"
+    let filter = try Filter(pattern: "^(.+\\/|)SG[^\\/]*\\.json$")
```

Comment:
Quite unusual in a RegEx to see `(x|)` to match either x or… nothing. We usually just use `x?` instead to match "maybe x"… So `^(.+/)?SG[^/]*\\.json$` should be enough (btw, why did you escape the `/` in those RegExes?)

Reply:
No idea why I escaped them 😆, it was late and I assumed it was some special character for some reason.

Reply:
I can't remember if it works correctly without the `(x|)`, will give it a quick test.